pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -161,6 +163,10 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 		var ok bool
 		newRc, ok = obj.(*api.ReplicationController)
 		if !ok {
+			if _, kind, err := typer.ObjectVersionAndKind(obj); err == nil {
+				return cmdutil.UsageError(cmd, "%s contains a %s not a ReplicationController", filename, kind)
+			}
+			glog.V(4).Infof("Object %#v is not a ReplicationController", obj)
```

Comment:
I guess it doesn't really matter.  It's the exception case of the exceptional case.